packages/framework/src/hooks/useMenuContext.tsx

```diff
@@ -0,0 +1,29 @@
+import type { PropsWithChildren } from "react";
+import React from "react";
+import { useHydrateAtoms } from "jotai/utils";
+import { menuImportScriptAtom } from "../state";
+
+interface MenuContextProps {
+  menuImportedScripts: string | undefined;
+}
+
+type MenuContextProviderProps = PropsWithChildren<MenuContextProps>;
+
+const HydrateMenuAtoms: React.FC<MenuContextProviderProps> = ({
+  menuImportedScripts,
+  children,
+}) => {
+  useHydrateAtoms([[menuImportScriptAtom, menuImportedScripts]]);
+  return <>{children}</>;
+};
+
+export const MenuContextProvider: React.FC<MenuContextProviderProps> = ({
```

Comment:
There's no need to create an entirely new provider. You can reuse the screen context provider. It's a bit hacky unless you want to refactor it but each each provider has its own scope for the atom value.